DevOps.Util/BaseServer.cs

```diff
@@ -0,0 +1,65 @@
+﻿#nullable enable
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace DevOps.Util
+{
+    public abstract class BaseServer
+    {
+        public IAzureClient AzureClient { get; }
```

Comment:
Why does the `HelixServer` need to use the `IAzureClient` here? My intuition is that it would only need essentially `HttpClient` and that `IAzureClient` would live in the `DevOpsServer` layer.

Reply:
I was trying to take advantage of the `GetJson*`, `DownloadFileAsync`, apis that are common in between both servers and it uses `AzureClient`: 
https://github.com/jaredpar/devops-util/pull/10/files#diff-7b795ac90edfe281b0422ebca80fadcdR37

It seems like `AzureClient` just do pure http requests using `HttpClient` and what is special about it is the azure authentication. So we could probably introduce an extra shared layer for non-authenticated clients vs authenticated clients.

Reply:
Also, HelixServer always create AzureClient with a null PAT, which means it’ll do regular not authenticated requests.